src/rpc/misc.cpp

```diff
@@ -597,6 +599,49 @@ static UniValue echo(const JSONRPCRequest& request)
     return request.params;
 }
 
+void AddSummary(IndexSummary&& summary, UniValue& results)
+{
+    UniValue entry(UniValue::VOBJ);
+    entry.pushKV("name", summary.name);
+    entry.pushKV("synced", summary.synced);
+    entry.pushKV("best_block_height", summary.best_block_height);
+    results.push_back(entry);
+}
+
+static UniValue listindices(const JSONRPCRequest& request)
+{
+            RPCHelpMan{"listindices",
+                "\nReturns the available indices and their status.\n",
+                {},
+                RPCResult{
+                    RPCResult::Type::ARR, "", "",
+                    {
+                        {RPCResult::Type::OBJ, "", "",
+                        {
+                            {RPCResult::Type::STR, "name", "The name of the index"},
+                            {RPCResult::Type::BOOL, "synced", "Whether the index is synced or not"},
+                            {RPCResult::Type::NUM, "best_block_height", "The block height to which the index is synced"},
+                        }},
+                    }
+                },
+                RPCExamples{
+                    HelpExampleCli("listindices", "")
+                },
+            }.Check(request);
+
+    UniValue results(UniValue::VARR);
+
+    if (g_txindex) {
+        AddSummary(g_txindex->GetSummary(), results);
```

Comment:
nit: Could avoid passing results around

```suggestion
        results.push_back(SummaryToJson(g_txindex->GetSummary()));
```